pymc3/distributions/discrete.py

```diff
@@ -285,8 +285,13 @@ def __init__(self, p, *args, **kwargs):
     def logp(self, value):
         p = self.p
         k = self.k
-
-        return bound(log(p[value]),
+        
+        try:
+            pv = log(p[value])
+        except IndexError:
+            pv = log(constant(p)[value])
```

Comment:
I think it would be better to use `as_tensor_variable` in the init. Constant can fail if it can't be converted to a constant if its already a tensor.